gtda/time_series/embedding.py

```diff
@@ -208,52 +213,169 @@ def plot(Xt, sample=0, plotly_params=None):
+    inputs are interpreted as collections of time series (one per entry along
+    axis 0), and the algorithm is applied to each time series separately to
+    produce a 3D array of point clouds. More generally, N-dimensional array
+    inputs are interpreted as containing one time series per entry along the
+    first N - 1 axes, and a (N + 1)-dimensional array of point clouds is
+    returned.
+
+    Parameters
+    ----------
+    X : ndarray of shape (n_timestamps,) or (n_time_series, ..., n_timestamps)
+        Input time series or collection of time series.
+
+    time_delay : int, optional, default: ``1``
+        Time delay between two consecutive values for constructing one embedded
+        point.
+
+    dimension : int, optional, default: ``2``
+        Dimension of the embedding space.
+
+    stride : int, optional, default: ``1``
+        Stride duration between two consecutive embedded points.
+
+    flatten : bool, optional, default: ``False``
+        If ``True``, and if ``X.ndim > 2``, ensures that the output is a 3D
+        ndarray of shape
+        ``(n_time_series, n_points, dimension * np.prod(X.shape[1:-1]))``,
+        where ``n_points`` is equal to
+        ``(n_timestamps - time_delay * (dimension - 1) - 1) // stride + 1``.
+
+    ensure_last_value : bool, optional, default: ``True``
+        Whether the value(s) of `X` representing the last measurement(s) must
+        be present in the output as the last coordinate(s) of the last
+        embedding vector(s). If ``False``, the first measurement(s) is (are)
+        present as the 0-th coordinate(s) of the 0-th vector(s) instead.
+
+    validate : bool, optional, default: ``True``
+        Whether keyword arguments should be validated.
+
+    Returns
+    -------
+    X_embedded : ndarray of shape (n_points, dimension) or \
+        (n_time_series, ..., n_points, dimension)
+        The result (a view on `X`) of the time-delay embedding of `X` with the
+        given parameters. If `X` is 1D, `X_embedded` is a single point cloud in
+        Euclidean space of dimension given by `dimension`. If `X` is
+        N-dimensional, `X_embedded` contains one such point cloud per entry of
+        `X` along its first N - 1 axes. In both cases, ``n_points`` is equal to
+        ``(n_timestamps - time_delay * (dimension - 1) - 1) // stride + 1``.
+
+    See also
+    --------
+    TakensEmbedding, SlidingWindow
+
+    Notes
+    -----
+    When `X` is 1D and `ensure_last_value` is ``True``, this function gives
+    identical outputs as the :meth:`fit_transform` method of a
+    :class:`TakensEmbedding` object instantiated with
+    ``parameters_type='fixed'`` (and its other parameters set appropriately).
+    However, :class:`TakensEmbedding` only accepts 2D array input when it is a
+    column vector, in which case it is flattened to 1D and interpreted as a
+    single time series.
+
+    """
+    if validate:
+        expected_params = {
```

Comment:
Where is `validate` validated ? xD

Maybe use something like:
```
expected_params = TakensEmbedding._hyperparameters.copy()
expected_params.pop('parameters_type')
expected_params.update(........)
```
To ensure that TakensEmbedding and this fonction are always maintained together (and to reduce number of lines a bit).

Reply:
I'll try something along those lines!

Reply:
Done, and `validate` is now also validated in the line above.